client/my-sites/hosting/sftp-card/index.js

```diff
@@ -19,18 +19,63 @@ import Button from 'components/button';
 import ClipboardButton from 'components/forms/clipboard-button';
 import Spinner from 'components/spinner';
 import { getSelectedSiteId } from 'state/ui/selectors';
+import { getCurrentUserId } from 'state/current-user/selectors';
 import {
-	requestAtomicSFTPDetails,
-	resetAtomicSFTPUserPassword,
-	createAtomicSFTPUser,
-} from 'state/data-getters';
+	requestAtomicSftpUser,
+	createAtomicSftpUser,
+	resetAtomicSftpPassword,
+	updateAtomicSftpUser,
+} from 'state/hosting/actions';
+import { getAtomicHostingSftpUser } from 'state/selectors/get-atomic-hosting-sftp-user';
 
-const SFTPCard = ( { translate, username, password, siteId, loading, disabled } ) => {
+const SftpCard = ( {
+	translate,
+	username,
+	password,
+	siteId,
+	disabled,
+	currentUserId,
+	requestSftpUser,
+	createSftpUser,
+	resetSftpPassword,
+	removePasswordFromState,
+} ) => {
 	// State for clipboard copy button for both username and password data
 	const [ isCopied, setIsCopied ] = useState( false );
+	const [ isLoading, setIsLoading ] = useState( false );
 	const usernameIsCopied = isCopied === 'username';
 	const passwordIsCopied = isCopied === 'password';
 
+	const onDestroy = () => {
+		if ( password ) {
+			removePasswordFromState( siteId, currentUserId, username );
+		}
+	};
+
+	const resetPassword = () => {
+		setIsLoading( true );
+		resetSftpPassword( siteId, currentUserId );
+	};
+
+	const createUser = () => {
+		setIsLoading( true );
+		createSftpUser( siteId, currentUserId );
+	};
+
+	useEffect( () => {
+		if ( ! username && ! disabled ) {
+			setIsLoading( true );
+			requestSftpUser( siteId, currentUserId );
+		}
+		return onDestroy();
```

Comment:
Since we run `onDestroy` every time `username` changes, the password is being removed from the state when creating a new SFTP user.

Reply:
haha - so that is why we had the loaded variable :-) I have reintroduced this but renamed it to  `sftpUserRequested` which I think is a bit clearer than `loaded` and this fixes the problem.